packages/studio/tailwind.config.ts

```diff
@@ -1,9 +1,60 @@
 import type { Config } from "tailwindcss";
+import {
+  StudioTailwindConfig,
+  StudioTailwindTheme,
+  STUDIO_PROCESS_ARGS_OBJ,
+  CliArgs,
+} from "@yext/studio-plugin";
 import path from "path";
+import fs from "fs";
+
+const getRootDir = (): string => {
+  const cliArgs: CliArgs = JSON.parse(
+    process.env[STUDIO_PROCESS_ARGS_OBJ] as string
+  );
+  return cliArgs.root ?? process.cwd();
+};
+const rootDir = getRootDir();
+
+/**
+ * The user's StudioTailwindTheme.
+ */
+const userTailwindTheme: StudioTailwindTheme =
+  (() => {
+    try {
+      const tailwindConfigPath = path.resolve(rootDir, "tailwind.config.ts");
+      if (fs.existsSync(tailwindConfigPath)) {
+        // We have to use require() instead of a dynamic import because
+        // tailwind does not support async config.
+        // eslint-disable-next-line @typescript-eslint/no-var-requires
+        const tailwindConfig: StudioTailwindConfig = require(tailwindConfigPath);
+        return tailwindConfig?.theme?.extend;
+      }
+    } catch (e) {
+      console.error(e);
+    }
+  })() ?? {};
+
+/**
+ * Generates a safelist for custom test colors, background colors, and font sizes.
+ */
+const generateSafelist = (theme: StudioTailwindTheme): string[] => {
+  const customColors = Object.keys(theme?.colors ?? {});
+  const customFontSizes = Object.keys(theme?.fontSize ?? {});
+  return [
+    ...customFontSizes.map((size) => `text-${size}`),
+    ...customColors.flatMap((color) => [`bg-${color}`, `text-${color}`]),
+  ];
+};
 
 export default {
-  content: [path.join(__dirname, "src/**/*.tsx")],
+  content: [
+    path.resolve(__dirname, "src/**/*.{ts,tsx}"),
+    path.resolve(__dirname, "index.html"),
+    path.resolve(rootDir, "src/**/*.{ts,tsx}"),
```

Comment:
For now, are we deliberately making the assumption that this is the only allowed content path for the user's Tailwind? Is there a reason for not including whatever content path/glob they have specified?

Reply:
We can't just directly spread the user's content array into ours because the relative paths won't work, and there's a few different ways to specify content besides just a string filepath
```ts
type ContentConfig =
  | (FilePath | RawFile)[]
  | {
      files: (FilePath | RawFile)[]
      relative?: boolean
      extract?: ExtractorFn | { [extension: string]: ExtractorFn }
      transform?: TransformerFn | { [extension: string]: TransformerFn }
    }
type FilePath = string
type RawFile = { raw: string; extension?: string }
```

But we can restrict what the user is allowed to specify as content via the StudioTailwindConfig type

Reply:
We still need to have a default here for when user's don't have a tailwind config. In the future maybe we can assume they always have one